lib/elixir_tic_tac_toe_basic/prompter.ex

```diff
@@ -0,0 +1,46 @@
+defmodule ElixirTicTacToeBasic.Prompter do
+  @moduledoc """
+  Prompts the user and returns valid input
+  """
+
+  def get_input(
+        state,
+        gets \\ fn state, prompt ->
```

Comment:
**question**: I am having trouble understanding this. Can you walk me through how this works?